samples/apps/copilot-chat-app/webapp/src/libs/useContentModerator.ts

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+import { useMsal } from '@azure/msal-react';
+import { useAppDispatch, useAppSelector } from '../redux/app/hooks';
+import { RootState } from '../redux/app/store';
+import { FeatureKeys } from '../redux/features/app/AppState';
+import { addAlert, toggleFeatureState } from '../redux/features/app/appSlice';
+import { AuthHelper } from './auth/AuthHelper';
+import { AlertType } from './models/AlertType';
+import { ContentModerationService } from './services/ContentModerationService';
+
+const riskThreshold = 4;
+
+export const useContentModerator = () => {
+    const dispatch = useAppDispatch();
+    const { inProgress, instance } = useMsal();
+    const { features } = useAppSelector((state: RootState) => state.app);
+
+    const contentModeratorService = new ContentModerationService(process.env.REACT_APP_BACKEND_URI as string);
+
+    const analyzeImage = async (base64Image: string) => {
+        const VIOLATIONS_FLAG = 'ContainsViolations';
+        try {
+            // remove image prefix
+            const image = base64Image.replace('data:image/png;base64,', '').replace('data:image/jpeg;base64,', '');
```

Comment:
a regex could be used here for brevity:
```suggestion
            const image = base64Image.replace(/data:image\/(png|jpeg);base64/, '');
```